app/models/material_batch.rb

```diff
@@ -22,13 +22,14 @@ def update_from_params(params)
       if material_update_params(params)[:relationships] and material_update_params(params)[:relationships][:materials]
         material_update_params(params)[:relationships][:materials][:data].each { |param|
           if param[:id]
-            material = Material.find(param[:id])
+            material = Material.find_by(uuid: param[:id])
             material.update_from_params(param)
 
             unless self.materials.include? material
-              self.materials << material
+              self.errors.add :materials, 'can\'t be added to an existing batch'
```

Comment:
We started to externalise the messages, so we could add this to the locales/en.yml, too.